multiqc/modules/biscuit/biscuit.py

```diff
@@ -64,214 +66,210 @@ def __init__(self):
 
         # Make report sections
-        for k in self.mdata:
-            if len(self.mdata[k]) > 0:
-                log.debug(f"Found {len(self.mdata[k])} {k} reports")
-                getattr(self, f"chart_{k}")()
+        if len(self.biscuit_data["align_mapq"]) > 0:
+            log.info(f"Found {len(self.biscuit_data['align_mapq'])} BISCUIT MAPQ reports")
+            self.chart_align_mapq()
+        if len(self.biscuit_data["align_strand"]) > 0:
+            log.info(f"Found {len(self.biscuit_data['align_strand'])} BISCUIT bisulfite strand reports")
+            self.chart_align_strand()
+        if len(self.biscuit_data["align_isize"]) > 0:
+            log.info(f"Found {len(self.biscuit_data['align_isize'])} BISCUIT insert size reports")
+            self.chart_align_isize()
+        if len(self.biscuit_data["dup_report"]) > 0:
+            log.info(f"Found {len(self.biscuit_data['dup_report'])} BISCUIT duplicate reports")
+            self.chart_dup_report()
+        if len(self.biscuit_data["qc_cv"]) > 0:
+            log.info(f"Found {len(self.biscuit_data['qc_cv'])} BISCUIT coefficient of variation reports")
+            self.chart_qc_cv()
+        if n_covdist_samples > 0:
+            log.info(f"Found {n_covdist_samples} BISCUIT coverage distribution reports")
+            self.chart_covdist()
+        if n_retention_readpos > 0:
+            log.info(f"Found {n_retention_readpos} BISCUIT retention by read position reports")
+            self.chart_retention_readpos()
+        if n_avg_retention > 0:
+            log.info(f"Found {n_avg_retention} BISCUIT average cytosine retention reports")
+            self.chart_avg_retention()
+
+        # Write data to file
+        self.write_data_file(self.biscuit_data, "biscuit", data_format="yaml")
 
     def biscuit_stats_table(self):
-        """
-        Create general statistics table for BISCUIT data
-        Inputs:
-            Uses mdata['align_mapq'] and mdata['dup_report']
-        Returns:
-            Add columns to MultiQC general statistics table
-        """
+        """BISCUIT data for the general statistics table"""
         pd = {}
 
-        # Calculate % aligned
-        for s_name, dd in self.mdata["align_mapq"].items():
-            if len(dd) > 0:
-                pd[s_name] = {"aligned": dd["frc_align"]}
+        # Calculate percent aligned
+        for s_name, data in self.biscuit_data["align_mapq"].items():
+            pd[s_name] = {"aligned": data["frac_align"]}
 
-        # Calculate % duplicated
-        for s_name, dd in self.mdata["dup_report"].items():
+        # Calculate percent duplicated
+        for s_name, data in self.biscuit_data["dup_report"].items():
             if s_name not in pd:
                 pd[s_name] = {}
-            if "all" in dd and dd["all"] != -1:
-                pd[s_name]["dup_all"] = dd["all"]
-            if "q40" in dd and dd["q40"] != -1:
-                pd[s_name]["dup_q40"] = dd["q40"]
```

Comment:
Presumably this `-1` was in the code for a reason, ie. it is sometimes found in the raw data. Is there a reason you removed it here?